src/components/figures/image/image.ts

```diff
@@ -0,0 +1,122 @@
+import { Component, useRef, useState } from "@odoo/owl";
+import { MENU_WIDTH } from "../../../constants";
+import { MenuItemRegistry } from "../../../registries/index";
+import { _lt } from "../../../translation";
+import { DOMCoordinates, Figure, SpreadsheetChildEnv, UID } from "../../../types";
+import { css } from "../../helpers";
+import { useAbsolutePosition } from "../../helpers/position_hook";
+import { Menu, MenuState } from "../../menu/menu";
+
+// -----------------------------------------------------------------------------
+// STYLE
+// -----------------------------------------------------------------------------
+css/* scss */ `
+  .o-image-menu {
+    right: 0px;
+    display: none;
+    position: absolute;
+    padding: 5px;
+  }
+  .o-figure.active:focus,
+  .o-figure:hover {
+    .o-image-menu {
+      display: flex;
+    }
+  }
+`;
+
+interface Props {
+  figure: Figure;
+}
+
+export class ImageFigure extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-ImageFigure";
+  static components = { Menu };
+  private menuState: MenuState = useState({ isOpen: false, position: null, menuItems: [] });
+
+  private imageContainerRef = useRef("o-image");
+  private menuButtonRef = useRef("menuButton");
+  private menuButtonPosition = useAbsolutePosition(this.menuButtonRef);
+  private position = useAbsolutePosition(this.imageContainerRef);
+
+  private getMenuItemRegistry(): MenuItemRegistry {
+    const registry = new MenuItemRegistry();
+    registry.add("copy", {
+      name: _lt("Copy"),
+      description: "Ctrl+C",
+      sequence: 1,
+      action: async () => {
+        this.env.model.dispatch("SELECT_FIGURE", { id: this.figureId });
+        this.env.model.dispatch("COPY");
+      },
+    });
+    registry.add("cut", {
+      name: _lt("Cut"),
+      description: "Ctrl+X",
+      sequence: 2,
+      action: async () => {
+        this.env.model.dispatch("SELECT_FIGURE", { id: this.figureId });
+        this.env.model.dispatch("CUT");
+      },
+    });
+    registry.add("reset_size", {
+      name: _lt("Reset size"),
+      sequence: 3,
+      action: async () => {
+        const update = this.env.model.getters.getImageSize(this.figureId);
+        this.env.model.dispatch("UPDATE_FIGURE", {
+          sheetId: this.env.model.getters.getActiveSheetId(),
+          id: this.props.figure.id,
+          ...update,
+        });
```

Comment:
Much more readable then an `update` object where you don't know what's inside (the naming is bad, "update" doesn't mean anything here)
```suggestion
        const { height, width } = this.env.model.getters.getImageSize(this.figureId);
        this.env.model.dispatch("UPDATE_FIGURE", {
          sheetId: this.env.model.getters.getActiveSheetId(),
          id: this.props.figure.id,
          height,
          width,
        });
```